docs-gb/kubernetes/server-autoscaling.md

```diff
@@ -0,0 +1,124 @@
+
+There is a period of time where the new server replicas are being created and the new model replicas are being loaded onto these server replicas. During this period, the system will ensure that the existing server replicas are still serving load so that there is no downtime during the scale up process. This is achieved by using partial scheduling of the new model replicas onto the new server replicas. This ensures that the new server replicas are gradually loaded with the new model replicas and that the existing server replicas are still serving load. Check the [Partial Scheduling](../models/scheduling.md) document for more details.
+
+## Server Scale Down
+
+### Overview
+
+When we want to scale down a model replicas, the associated servers might be left unused. In this case, extra server pods (especially if they use GPUs) are wasted causing infrastructure cost for the user.
+
+Scaling down servers in sync with models is not straight forward in the case of multi-model serving. Scaling down one model does not necessarily mean that we also need to scale down the corresponding server replica as this server replica might be still serving load for other models. 
+
+Therefore we need to define some heuristics that can be used to scale down servers if we think that they are not properly used.
+
+{% hint style="info" %}
+**Note**: Scaling down the number of replicas for a model server does not necessarily mean that the system is going to remove a specific replica that we want. 
+
+As currently we have the model server deployed as StatefulSets, scaling down the number of replicas will mean that we are removing a pod with the largest index.
+
+The system will rebalance afterwards, where the models from this draining server replica will be rescheduled but this requires some wait time to achieve. This draining process should allow for no downtime as models are being rescheduled onto other server replicas before the draining server replica is removed.
+{% endhint %}
+
+### Policies
+
+1. **Empty Server Replica**:
+    
+In the simplest case we can remove a server replica if it does not host any models. This guarantees that there is no load on a particular server replica before removing it.
+
+This policy works best in the case of single model serving where the server replicas are only hosting a single model. In this case, if the model is scaled down, the server replica will be empty and can be removed.
+
+However in the case of MMS it can lead to less optimal packing of models onto server replicas. This is because the system will not automatically pack models onto the smaller set of replicas. This can lead to more server replicas being used than necessary. This can be mitigated by the lightly loaded server replicas policy.    
```

Comment:
```suggestion
However in the case of MMS, only reducing the number of server replicas when a server no longer hosts any models can lead to a suboptimal packing of models onto server replicas. This is because the system will not automatically pack models onto the smaller set of replicas. This can lead to more server replicas being used than necessary. This can be mitigated by the lightly loaded server replicas policy.    
```